### PR TITLE
 added addresses model with association to customer and business and …

### PR DESCRIPTION
**Pull request by nooralnabi**
…add google map

### CODE REVIEW

File: app/views/addresses/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,67 @@
+<%= form_with(model: address, local: true) do |form| %>
```
Comment (ssharba) [inline app/views/addresses/_form.html.erb]: haml

File: app/controllers/addresses_controller.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  def index
+    @addresses = Address.all
+  end
+
+
+  def map
+    @addresses = Address.all
+    @unique_state = Address.pluck(:state).uniq.compact
+    @customer_by_id_state = Address.customer_by_id(params[:selected_state])
+
```
Comment (MwMaki) [inline app/controllers/addresses_controller.rb]: please remove  empty lines

File: app/models/address.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  # belongs_to :owner, polymorphic: true
+
+  geocoded_by :address
+  after_validation :geocode, if: :address_changed?
+
+  # validates_presence_of :latitude, message: "Please check your address"
+   # validates :longitude, message: "Please check your address"
+   # validates :longitude, numericality: { greater_than_or_equal_to: -90,  \
+   #   less_than_or_equal_to: 90 }
+ STATES = ['Baghdad', 'Babil', 'Dohuk', 'Nineveh', 'Erbil','Kirkuk', 'Sulaymaniyah', 'Saladin', 'Al Anbar', 'Diyala', 'Karbala', 'Wasit', 'Najaf', 'Al-Qādisiyyah', 'Maysan', 'Dhi Qar', 'Basra']
```
Comment (MwMaki) [inline app/models/address.rb]: don't use upper case variables please

File: app/views/addresses/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<p id="notice"><%= notice %></p>
```
Comment (MwMaki) [inline app/views/addresses/index.html.erb]: haml please

File: app/models/province.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class Province < ApplicationRecord
+  has_many :cities
+  has_many :addresses
```
Comment (MwMaki) [inline app/models/province.rb]: @nooralnabi 
if cities belongs_to Province
and address belongs_to cities 
then 
why we need association between Province and address?
why you didn't define it as  Province has_many addresses through  city ?

**Pull request merged**